src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.Simple.cs

```diff
@@ -57,6 +57,10 @@ public void WriteDouble(double value)
         }
         private void WriteSingleCore(float value)
         {
+            if (float.IsNaN(value))
+            {
+                value = CborHelpers.Int32BitsToSingle(0x7fc00000); // canonical NaN as per RFC 7049
```

Comment:
could these use constants? CanonicalFloat defined here or just float.NaN or something? `CborHelpers.Int32BitsToSingle(0x7fc00000);` and `BitConverter.Int64BitsToDouble(0x7ff8000000000000)` seem constant or am I missing something?

Reply:
Sure, I put them in named constants.

`float.NaN` is defined as `0.0f / 0.0f` which doesn't yield a consistent bit pattern across platforms (it's negative on x86, positive elsewhere).